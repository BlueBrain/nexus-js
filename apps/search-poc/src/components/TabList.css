.tab-item {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  text-align: left;
  max-width: 250px;
}

.tab-item > .title {
  /* font-size: 1em; */
  font-weight: bold;
}

.tab-item > .description {
  white-space: normal;
  line-height: 1.2em;
  color: rgba(0, 0, 0, 0.5);
}
