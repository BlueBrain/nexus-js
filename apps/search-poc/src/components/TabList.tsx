import * as React from 'react';
import './TabList.css';
import { Tabs } from 'antd';

const { TabPane } = Tabs;

type WorkspaceItem = {
  id: string;
  label: string;
  description: string;
};

type TabListProps = {
  items: WorkspaceItem[];
  onSelected: (activeKey: string) => void;
  defaultActiveId?: string;
  position?: string;
};

const TabList: React.FunctionComponent<TabListProps> = ({
  items,
  onSelected,
  defaultActiveId,
  position = 'left',
  children,
}) => {
  return (
    <div className="tab-list">
      <Tabs
        defaultActiveKey={defaultActiveId}
        tabPosition={position}
        onChange={onSelected}
      >
        {items.map(({ label, description, id }) => (
          <TabPane
            tab={
              <div className="tab-item">
                <span className="title ellipsis">{label}</span>
                <p className="description fade">{description}</p>
              </div>
            }
            key={id}
          >
            {children}
          </TabPane>
        ))}
      </Tabs>
    </div>
  );
};

export default TabList;
