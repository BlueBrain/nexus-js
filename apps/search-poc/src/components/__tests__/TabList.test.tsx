import * as renderer from "react-test-renderer";
import React from "react";
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils';
import TabList from '../TabList.tsx';
import { shallow } from "enzyme";


it("TabList renders correctly with work items", () => {
  const props = { items: [{id: 'id', onSelected: () => {}, defaultActiveId: '1'}], onSelected: jest.fn};
  
  const result = shallow(<TabList {...props} />);
  expect(result).toMatchSnapshot();
});