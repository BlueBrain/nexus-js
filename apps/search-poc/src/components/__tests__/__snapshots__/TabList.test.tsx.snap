// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabList renders correctly with work items 1`] = `
<div
  className="tab-list"
>
  <Tabs
    hideAdd={false}
    onChange={[Function]}
    tabPosition="left"
  >
    <TabPane
      key="id"
      placeholder={null}
      tab={
        <div
          className="tab-item"
        >
          <span
            className="title ellipsis"
          />
          <p
            className="description fade"
          />
        </div>
      }
    />
  </Tabs>
</div>
`;
